pkg/diff/diff.go

```diff
@@ -45,6 +56,49 @@ type EntityChanges struct {
 	Deleting []EntityState `json:"deleting"`
 }
 
+// ------------------------------------------------------
+// New types used by the no output diff engine
+// ------------------------------------------------------
+
+// ReconcileAction is an action taken by the diff engine.
+type ReconcileAction string
+
+const (
+	// CreateAction is the ReconcileAction used when a target object did not exist in the current state and was created.
+	CreateAction = ReconcileAction("create")
+	// UpdateAction is the ReconcileAction used when a target object exists in the current state and was updated.
+	UpdateAction = ReconcileAction("update")
+	// DeleteAction is the ReconcileAction used when a current object exists in the target state and was deleted.
+	DeleteAction = ReconcileAction("delete")
+
+	// eventBuffer is the number of events to buffer in the various syncer channels.
+	eventBuffer = 10
```

Comment:
Is there any reason for setting `eventBuffer` to be this constant?

Reply:
Not really. It was an existing value from https://github.com/Kong/go-database-reconciler/pull/76/files#diff-00282c65a618a9ea64cdb99da5137663dc5773f2c3fd8c37ed2e9a99f3d67f09L347; I just moved the definition.

I expect that it's relatively small to avoid overloading the admin API with a ton of requests in flight if the DB is slow to complete them. Could make sense to make it configurable (a library const isn't great for our new client/library model), but we haven't found that we need to configure it so far.